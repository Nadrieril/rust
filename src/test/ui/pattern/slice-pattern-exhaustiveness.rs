#![feature(slice_patterns)]

fn main() {
    let s: [bool; 1] = [false; 1];
    match s {
        [a] => {}
    }
    match s {
        [a, ..] => {}
    }
    match s {
        [true, ..] => {}
        [.., false] => {}
    }

    let s: [bool; 2] = [false; 2];
    match s {
        [a, b] => {}
    }
    match s {
        [a, ..] => {}
    }
    match s {
    //~^ ERROR `[false, true]` not covered
        [true, ..] => {}
        [.., false] => {}
    }
}
