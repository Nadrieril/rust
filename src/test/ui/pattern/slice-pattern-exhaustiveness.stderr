error[E0004]: non-exhaustive patterns: `[false, true]` not covered
  --> $DIR/slice-pattern-exhaustiveness.rs:23:11
   |
LL |     match s {
   |           ^ pattern `[false, true]` not covered
   |
   = help: ensure that all possible cases are being handled, possibly by adding wildcards or more match arms

error: aborting due to previous error

For more information about this error, try `rustc --explain E0004`.
